src/printer.js

```diff
@@ -1729,6 +1729,31 @@ function genericPrintNoParens(path, options, print, args) {
     case "TemplateElement":
       return join(literalline, n.value.raw.split(/\r?\n/g));
     case "TemplateLiteral": {
+      /* Subtree parsing PoC */
+      const parent = path.getParentNode();
+      const parentParent = path.getParentNode(1);
+      const isStyledJsx =
+        n.quasis &&
+        n.quasis.length === 1 &&
+        parent.type === "JSXExpressionContainer" &&
+        parentParent.type === "JSXElement" &&
+        parentParent.openingElement.attributes.some(
+          attribute => attribute.name.name === "jsx"
+        );
+
+      if (isStyledJsx) {
+        const parseCss = require("./parser-postcss");
+
+        const text = n.quasis[0].value.raw;
+        const ast = parseCss(text, options);
+        let subtree = printAstToDoc(ast, options);
+        //HACK remove ending hardline
+        subtree = subtree.parts[0].parts[0];
```

Comment:
Looks like it's likely not going to be trivial for the js part, the hardline is now always printed at the top level.